File: CS/Q157926/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraLayout.Helpers;
using DevExpress.XtraLayout;
using System.Collections;
using DevExpress.XtraEditors;

namespace Q157926 {
    public partial class Form1 :Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            // TODO: This line of code loads data into the 'nwindDataSet.Products' table. You can move, or remove it, as needed.
            this.productsTableAdapter.Fill(this.nwindDataSet.Products);
            layout.FocusHelper = new MyFocusHelper(layout);
        }
    }

    public class MyFocusHelper :FocusHelperBase {
        public MyFocusHelper(ILayoutControl control) : base(control) { }

        private LayoutGroup focusedGroup;
        protected override ArrayList FilterElements(ArrayList list, bool tabStopOnly) {
            ContainerControl container = layoutControl.Control as ContainerControl;
            if (container != null && container.ActiveControl != null) {
                Control activeControl = container.ActiveControl;
                if (activeControl is TextBoxMaskBox) activeControl = activeControl.Parent;
                focusedGroup = layoutControl.GetItemByControl(activeControl).Parent;
            }
            //focusedGroup = null;
            return base.FilterElements(list, tabStopOnly);
        }

        protected override bool CheckItem(BaseLayoutItem item, bool tabStopOnly) {
            if (focusedGroup != null && item.Parent != focusedGroup) return false;
            return base.CheckItem(item, tabStopOnly);
        }
    }
}

File: VB/Q157926/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraLayout.Helpers
Imports DevExpress.XtraLayout
Imports System.Collections
Imports DevExpress.XtraEditors

Namespace Q157926
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'nwindDataSet.Products' table. You can move, or remove it, as needed.
			Me.productsTableAdapter.Fill(Me.nwindDataSet.Products)
			layout.FocusHelper = New MyFocusHelper(layout)
		End Sub
	End Class

	Public Class MyFocusHelper
		Inherits FocusHelperBase
		Public Sub New(ByVal control As ILayoutControl)
			MyBase.New(control)
		End Sub

		Private focusedGroup As LayoutGroup
		Protected Overrides Function FilterElements(ByVal list As ArrayList, ByVal tabStopOnly As Boolean) As ArrayList
			Dim container As ContainerControl = TryCast(layoutControl.Control, ContainerControl)
			If container IsNot Nothing AndAlso container.ActiveControl IsNot Nothing Then
				Dim activeControl As Control = container.ActiveControl
				If TypeOf activeControl Is TextBoxMaskBox Then
					activeControl = activeControl.Parent
				End If
				focusedGroup = layoutControl.GetItemByControl(activeControl).Parent
			End If
			'focusedGroup = null;
			Return MyBase.FilterElements(list, tabStopOnly)
		End Function

		Protected Overrides Function CheckItem(ByVal item As BaseLayoutItem, ByVal tabStopOnly As Boolean) As Boolean
			If focusedGroup IsNot Nothing AndAlso item.Parent IsNot focusedGroup Then
				Return False
			End If
			Return MyBase.CheckItem(item, tabStopOnly)
		End Function
	End Class
End Namespace